protocol/__init__.py

```diff
@@ -167,8 +173,35 @@ def create_file_hash(self, data):
         """
         sha3_hash = None
         with open(data, 'rb') as file_contents:
-            file_contents.read()
-            sha3_hash = self.w3.sha3(file_contents)
+            b = file_contents.read(1024*1024) # read file in 1MB chunks
+            while b:
+                sha3_hash = self.w3.sha3(b)
+                b = file_contents.read(1024*1024)
         return sha3_hash
 
+    def validate_candidate(self, listing_hash):
+        """
+        Verify a candidate has been submitted to Computable Protocol
+        by parsing logs for 'CandidateAdded' event and matching listing hash
+        Verify voteBy has not expired
+        :params listing_hash: String listing hash for listing
+        :return owner address if valid otherwise return None:
+        :return type string:
+        """
+        owner = None
+        voting_filter = self.voting.deployed.eventFilter(
+            constants.CANDIDATE_ADDED,{'fromBlock':0,'toBlock':'latest'}
+        )
+        events = voting_filter.get_all_entries()
+        for evt in events:
+            event_hash = '0x' + evt['args']['hash'].hex()
+            print(f'Comparing event hash {event_hash} to listing hash {listing_hash}')
+            if event_hash == listing_hash:
+                log.info(f'Listing {listing_hash} has been listed as a candidate in protocol')
+                voteBy = evt['args']['voteBy']
+                if voteBy > int(time.time()):
+                    owner = evt['args']['owner']
+        return owner
```

Comment:
once we recieve the `transaction_hash` from the client we can do 2 things:

1. use web3.py to `wait_for_transaciton_receipt` (that way we know its mined)
2. use protocol's `voting_contract.is_candidate(hash)`

Reply:
one question this raises is 

_should we implement twisted/celery/whatever to handle doing this in threads?_